core/wire/kem_test.go

```diff
@@ -0,0 +1,175 @@
+// kem_test.go - Wire protocol session KEM interfaces tests.
+// Copyright (C) 2022  David Anthony Stainton
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package wire implements the Katzenpost wire protocol.
+package wire
+
+import (
+	"encoding/base64"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/katzenpost/katzenpost/core/crypto/rand"
+)
+
+func TestSignatureScheme(t *testing.T) {
+	privKey1 := DefaultScheme.GenerateKeypair(rand.Reader)
+	pubKey1 := privKey1.PublicKey()
+
+	pubKey2, err := DefaultScheme.UnmarshalBinaryPublicKey(pubKey1.Bytes())
+	require.NoError(t, err)
+	require.True(t, pubKey1.Equal(pubKey2))
+
+	pubText, err := pubKey1.MarshalText()
+	require.NoError(t, err)
+	pubKey3, err := DefaultScheme.UnmarshalTextPublicKey(pubText)
+	require.NoError(t, err)
+	require.True(t, pubKey1.Equal(pubKey3))
+
+	_, err = DefaultScheme.PublicKeyFromPemFile("definitelydoesnotexist")
+	require.Error(t, err)
+
+	wrongPemData := `-----BEGIN ED25519 SPHINCS+ PUBLIC KEY-----
+2JQzwEwGBxBkQ0quWab4MD2T3E/WozBsfAMzp9wDLDaSm4jMADucY0gHAxX08iM3
+e/o5l00d9jhM5Gr51yY5FT8acP8IdPeDS1ccwW1HTpmAWMQOJyZwvo9jwiog9IVq
+-----END ED25519 SPHINCS+ PUBLIC KEY-----
+`
+	wrongPemPath := filepath.Join(os.TempDir(), "wrongpubkey1.pem")
```

Comment:
instead of `os.TempDir()` I think we want to use a single call to this [testing.TempDir()](https://pkg.go.dev/testing#T.TempDir) per unique directory we need. That way it will get deleted after the test runs, so we don't litter `/tmp` with tons of stuff.